### PR TITLE
Support stop current running job in job manager.

### PR DESCRIPTION
**Pull request by liuyu85cn**
as title

### CODE REVIEW

**Review by critical27 — APPROVED** [review]
LGTM

File: src/meta/processors/jobMan/MetaJobExecutor.cpp [hunk 1]
```diff
@@ -131,6 +131,17 @@ ErrOrHosts MetaJobExecutor::getLeaderHost(GraphSpaceID space) {
     return hosts;
 }
 
+void MetaJobExecutor::interruptExecution(JobID jobId) {
+    if (jobId == jobId_) {
+        std::unique_lock<std::mutex> lk(muInterrupt_);
+        stopped_ = true;
+        condInterrupt_.notify_one();
+        lk.unlock();
```
Comment (bright-starry-sky) [inline src/meta/processors/jobMan/MetaJobExecutor.cpp:139]: A question, Is this line of code redundant?

**Review by bright-starry-sky — APPROVED** [review]
Well done. nice !

**Review by panda-sheep — APPROVED** [review]

File: src/meta/processors/jobMan/MetaJobExecutor.cpp (hunk 1 above)
Comment (liuyu85cn) [reply src/meta/processors/jobMan/MetaJobExecutor.cpp:139]: just want to release lock , before that below LOG(INFO)

File: src/meta/processors/jobMan/MetaJobExecutor.cpp [hunk 2]
```diff
@@ -145,42 +156,32 @@ ExecuteRet MetaJobExecutor::execute() {
...
-                    LOG(ERROR) << "Admin Failed: " << t.exception();
-                    results.emplace_back(nebula::Status::Error());
-                } else {
-                    results.emplace_back(t.value());
-                }
+        std::move(fut).thenTry([&](auto&& t) {
+            std::lock_guard<std::mutex> lk(muInterrupt_);
+            --unfinishedTask;
+            if (t.hasValue()) {
+                ret[host] = t.value();
```
Comment (dutor) [inline src/meta/processors/jobMan/MetaJobExecutor.cpp:174]: There seems race condition on `ret`, involving line 167 vs. 174

File: src/meta/processors/jobMan/MetaJobExecutor.cpp [hunk 3]
```diff
@@ -145,42 +156,32 @@ ExecuteRet MetaJobExecutor::execute() {
...
+            condInterrupt_.notify_one();
+        });
     }
 
-    std::unordered_map<HostAddr, Status> ret;
-    for (size_t i = 0; i != addresses.size(); ++i) {
-        ret.emplace(addresses[i].first, results[i]);
+    {
+        std::unique_lock<std::mutex> lk(muInterrupt_);
+        condInterrupt_.wait(lk, [&]{ return unfinishedTask == 0 || stopped_; });
```
Comment (dutor) [inline src/meta/processors/jobMan/MetaJobExecutor.cpp:182]: I wonder in what context this sync operation gets executed. Will it block the normal bussiness logics?

File: src/meta/processors/jobMan/MetaJobExecutor.cpp (hunk 2 above)
Comment (liuyu85cn) [reply src/meta/processors/jobMan/MetaJobExecutor.cpp:174]: good catch, will fix

File: src/meta/processors/jobMan/MetaJobExecutor.cpp (hunk 3 above)
Comment (liuyu85cn) [reply src/meta/processors/jobMan/MetaJobExecutor.cpp:182]: job mgr run job one by one, and this where the current running job waiting for its sub task return.

**Review by dutor — APPROVED** [review]
LGTM.

**Review by critical27 — APPROVED** [review]
Good job

**Review by panda-sheep — APPROVED** [review]

**Pull request merged**